### PR TITLE
Various cleanups, preparations for updated android support

### PR DESCRIPTION
**Pull request by Vogtinator**
Builds against NDK r21 and Qt 5.14.1 for armeabi-v7a and arm64-v8a now.

I was surprised that clang as provided by the NDK doesn't understand the `@PAGE` syntax needed for clang for iOS, maybe that changed in a newer llvm version?

### CODE REVIEW

**Comment by Vogtinator** [comment]
I moved the android specific parts into #172 now.

File: firebird.pro [hunk 1]
```diff
@@ -43,6 +43,8 @@ LIBS += -lz
 # Override bad default options to enable better optimizations
 QMAKE_CFLAGS_RELEASE = -O3 -DNDEBUG
 QMAKE_CXXFLAGS_RELEASE = -O3 -DNDEBUG
+# I don't know why g++-unix.conf sets -Wl,-O1, override that.
+QMAKE_LFLAGS_RELEASE += -Wl,-O3
```
Comment (adriweb) [inline firebird.pro]: That needs a `!clang: ` as clang doesn't know that flag.

**Comment by Vogtinator** [comment]
> On a normal macOS build, I'm getting "Could not init JIT, disabling translation." (in both debug and release).

That means `os_alloc_executable` fails - maybe R^X is enforced on `macOS` as well now?

**Comment by adriweb** [comment]
So it turns out I have the same issue on master for some reason (definitely needs investigation), so otherwise I guess this PR is OK.

**Comment by adriweb** [comment]
OK, for JIT to work again, we need to not use `MAP_32BIT` anymore. See https://github.com/hrydgard/ppsspp/commit/d9b6ab83cfc0fe0225a00e146c2d1bd59c5467a4 for instance

**Comment by Vogtinator** [comment]
If you just remove `MAP_32BIT`, does it work? If not, the address hint is ignored and as the JIT does not support far calls it would crash.

**Comment by adriweb** [comment]
Yes, works fine when I just redefine MAP_32BIT to 0 if __APPLE__.

**Comment by Vogtinator** [comment]
It might not be needed on other platforms either - can you remove it and instead add a check that the returned address is below 2GiB?

**Comment by adriweb** [comment]
Well, to keep the same behavior as before, I suppose the best Way would be to only change that on macOS 10.15+ though.

**Comment by Vogtinator** [comment]
That would have less risk for regressions, so is the better option for now. Can you make a PR? I wouldn't be able to test it anyway...

**Pull request merged**